obs/project.go

```diff
@@ -91,3 +93,58 @@ type RepositoryPath struct {
 	Project    string `json:"project" xml:"project,attr"`
 	Repository string `json:"repository" xml:"repository,attr"`
 }
+
+type Status struct {
+	XMLName xml.Name `json:"status" xml:"status"`
+	Code    string   `json:"code" xml:"code,attr"`
+	Summary string   `json:"summary" xml:"summary"`
+}
+
+type APIError struct {
+	HTTPStatusCode int
+	XMLStatusCode  string
+	Message        string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("HTTP status %d: %s (%s)", e.HTTPStatusCode, e.XMLStatusCode, e.Message)
+}
+
+func (c Client) DeleteProject(project *Project) error {
+	client := &http.Client{}
+
+	req, err := http.NewRequest("DELETE", c.APIURL+"/source/"+project.Name, nil)
```

Comment:
This way of joining URLs is a bit error prone. I recommend taking a look at some answers on StackOverflow for an idea how to do it in a more proper way: https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join